src/lib/djerba/plugins/fusion/plugin.py

```diff
@@ -63,12 +67,129 @@ def sort_by_actionable_level(row):
             }
             gene_info = []
             treatment_opts = []
+
+        # Processing fusions and generating blob URLs
+        tsv_file_path = wrapper.get_my_string(fc.ARRIBA_PATH)
+        base_dir = (directory_finder(self.log_level, self.log_path).get_base_dir())
+        fusion_dir = os.path.join(base_dir, "plugins/fusion")
+        json_template_path = os.path.join(fusion_dir, fc.JSON_TO_BE_COMPRESSED)
+        output_dir = self.workspace.get_work_dir()
+        unique_fusions = list({item["fusion"] for item in results[fc.BODY]})
+        fusion_url_pairs = []
+
+        for fusion in unique_fusions:
+            try:
+                fusion, blurb_url = self.process_fusion(config, fusion, tsv_file_path, json_template_path, output_dir)
+                fusion_url_pairs.append([fusion, blurb_url])
+
+            except ValueError as e:
+                self.logger.error(f"Error processing fusion {fusion}: {e}")
+
+        # Save the fusion-URL pairs to a CSV file
+        output_tsv_path = os.path.join(output_dir, 'fusion_blurb_urls.tsv')
+        with open(output_tsv_path, 'w', newline='') as tsvfile:
+            writer = csv.writer(tsvfile, delimiter='\t')
+            writer.writerow(['Fusion', 'Whizbam URL'])
+            writer.writerows(fusion_url_pairs)
+
         data = self.get_starting_plugin_data(wrapper, self.PLUGIN_VERSION)
         data[core_constants.RESULTS] = results
         data[core_constants.MERGE_INPUTS]['gene_information_merger'] = gene_info
         data[core_constants.MERGE_INPUTS]['treatment_options_merger'] = treatment_opts
         return data
 
+    def process_fusion(self, config, fusion, tsv_file_path, json_template_path, output_dir):
+        base_dir = (directory_finder(self.log_level, self.log_path).get_base_dir())
+        wrapper = self.get_config_wrapper(config)
+        match = re.match(r"(.+)::(.+)", fusion)
+        if match:
+            gene1 = match.group(1)
+            gene2 = match.group(2)
+        else:
+            raise ValueError(f"No valid fusion found for {fusion}.")
+
+        # Initialize breakpoints
+        breakpoint1, breakpoint2 = None, None
+
+        # Open and read the ARRIBA TSV file
+        with open(tsv_file_path, mode='r') as file:
+            reader = csv.DictReader(file, delimiter='\t')
+
+            # Find the correct row based on gene1 and gene2
+            for row in reader:
+                if (row['#gene1'] == gene1 or row['#gene1'] == gene2) and (
+                        row['gene2'] == gene1 or row['gene2'] == gene2):
+                    breakpoint1 = row['breakpoint1']
+                    breakpoint2 = row['breakpoint2']
+                    break
+
+        if not (breakpoint1 and breakpoint2):
+            raise ValueError(f"No matching fusion found in the TSV file for {fusion}.")
```

Comment:
Should this be a fatal error? If so it should be logged. See comment on line 162.

Reply:
I added a try except block under the extract function to ensure that if a ValueError is raised for a particular fusion, it is caught by the except block and increments failed_fusions, then the code continues for the next fusion to be processed

Reply:
The problem with using `ValueError` is that it can be raised by all sorts of things in Python.
Raising an error and then catching it is OK, but in that case we should:

1. Log the error at the location where it is raised, in case needed to assist in debugging
2. Use a custom error class so we know exactly what kind of error it is -- here is an example of an [error class](https://github.com/oicr-gsi/djerba/blob/cdaea31c8f069f36b4b4a45b059a7bc899a8d3a9/src/lib/djerba/helpers/provenance_helper/helper.py#L284) and its [usage](https://github.com/oicr-gsi/djerba/blob/cdaea31c8f069f36b4b4a45b059a7bc899a8d3a9/src/lib/djerba/helpers/provenance_helper/helper.py#L7).

Reply:
Added a custom error class